### PR TITLE
Support generating bash completion

### PR DESCRIPTION
**Pull request by nak3**
This patch adds subcommand to generate completion script for bash.

As an knative user, we would like to complete subcommands and
options rather than typing them.

Usage Example:

```
$ source <(kn completion)

$ kn <TAB>
completion  revision    service

$ kn service list --<TAB>
--allow-missing-template-keys  --kubeconfig=                  --output=
--config=                      --namespace=                   --template=
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,3 +27,22 @@ $ go build ./cmd/...
...
+Run following to enable bash auto completion
+
+```sh
+$ source <(kn completion)
+```
+
+Use TAB to list available sub-commands
+
+```sh
+$ source <TAB>
```
Comment (navidshaikh) [inline README.md]: There is a typo @nak3

File: README.md [hunk 2]
```diff
@@ -27,3 +27,22 @@ $ go build ./cmd/...
...
+completion revision service
+
+$ source revision <TAB>
```
Comment (navidshaikh) [inline README.md]: yet another one. Sending patch to update this against your branch @nak3

**Comment by cppforlife** [comment]
/lgtm
/approve

not sure if cla check needs fixing?

**Comment by nak3** [comment]
Thank you @cppforlife Fixed CLA issue. Although LGTM label has been dropped, there are no code change.

**Pull request merged**